### ISSUE TITLE
I'd like to discuss Webpack

### ISSUE THREAD

**Issue by ghost** [ember-cli/rfcs#79]
With all the new modularization happening I'd like to champion swapping Broccoli out for Webpack in Ember CLI.

Without writing a big wall of text, here's my short outline of why:
- Broccoli is (pretty much) exclusive to Ember.
- Vue, Angular, React and a good few independant projects use Webpack, I've tried some of these in the past week and it's suprisingly easy to add dependencies.
- Webpack will give Ember CLI better support for generic packages. I think this will become the key pain point of keeping Broccoli around.
- Webpack has better documentation on writing plugins for webpack, this is pretty much undocumented in Broccoli's case.
- I'm sure we can make it work for existing pure ember (non-'cli') addons, without breaking much. I guess we'll need a transition period for the addons  that do, although some will become obsolete too.

TL;DR I think it is time to admit Broccoli is a dead end street and that Webpack will bring more benefits to Ember.

**Comment by stefanpenner** [comment]
> I'm sure we can make it work for existing pure ember (non-'cli') addons, without breaking much. I guess we'll need a transition period for the addons that do, although some will become obsolete too.

this is not the case

> Webpack will give Ember CLI better support for generic packages. I think this will become the key pain point of keeping Broccoli around.

it is not an either or, the worlds are different but a bridge can exist. We have been discussing webpack as a bundling step/terminal node \w the webpack team.

> Webpack has better documentation on writing plugins for webpack, this is pretty much undocumented in Broccoli's case.

effort is starting up to address this. cc @nathanhammond

**Comment by stefanpenner** [comment]
Just a heads-up, I do not subscribe chasing silver bullets. However large or small they seem, jumping around chasing some silver bullet rather then explicitly targeting and addressing issues yields to the fun "churn fatigue" many have identified.

A way to combat this, is to start with problems rather then solutions. Considering the existing state, explore how a given problem can be addressed. A good system works, by striving to incrementally remove issues and misalignments, learning along the way. This has been a guiding principle for this project, and that of ember.

**Now it is very important to understand deficiencies**, learn from competition, steal when appropriate, share when appropriate. The above described issues are real, but not unreasonable to address and improve in a reasonable timeframe, and as mentioned some of the efforts are complementary and should be based on shared efforts.

We had a great session with @TheLarkInn (of webpack) last week, and have some broad stroke ideas on what could exist.

The build system as a program environment, which is broccoli has some rough edges for-sure, but it is also a system that provides both leverage and aligns with both our short and long term goals. It is absolutely not perfect, but I do firmly believe that addressing those deficiencies (As we have been with every aspect of our offering) continues to be the direction that yields us the dividends we want.

---

Now let me encourage taking the time, to describe the problems in ways that we can collectively work to address.

**Comment by TheLarkInn** [comment]
Indeed. As the days draw closer to us releasing webpack 2 from beta, the closer we get to (after release) adding the optimization features that we discussed ( a combination of rollup style module inlining etc. + some generic parser based plugins that could take advantage of engine heuristics and micro-optimizations (where appropriate of course)). If we don't already have a tracking issue, we should get one going, so that we can document for the community any high level plans etc. Thanks for your interest in this @martndemus.

**Comment by rwjblue** [comment]
We are working on closing the ember-cli/rfcs repo in favor of using a single central RFC's repo for everything. This was laid out in https://emberjs.github.io/rfcs/0300-rfc-process-update.html. 

Sorry for the troubles, but would you mind reviewing to see if this is still something we need, and if so migrating this over to emberjs/rfcs?
